packages/core/modules-sdk/src/loaders/utils/load-internal.ts

```diff
@@ -155,8 +173,229 @@ export async function loadModuleMigrations(
+      importAllFromDir(resolve(normalizedPath, "dist", "services")).catch(
+        defaultOnFail
+      ),
+      importAllFromDir(resolve(normalizedPath, "dist", "models")).catch(
+        defaultOnFail
+      ),
+      importAllFromDir(resolve(normalizedPath, "dist", "repositories")).catch(
+        defaultOnFail
+      ),
+    ])
+
+    const cleanupResources = (resources) => {
+      return Object.values(resources).filter(
+        (resource): resource is Function => {
+          return typeof resource === "function"
+        }
+      )
+    }
+
+    const potentialServices = [...new Set(cleanupResources(services))]
+    const potentialModels = [...new Set(cleanupResources(models))]
+    const potentialRepositories = [...new Set(cleanupResources(repositories))]
+
+    const finalLoaders = prepareLoaders({
+      loadedModuleLoaders,
+      models: potentialModels,
+      repositories: potentialRepositories,
+      services: potentialServices,
+      moduleResolution,
+      migrationPath: normalizedPath + "/dist/migrations",
+    })
+
+    return {
+      services: potentialServices,
+      models: potentialModels,
+      repositories: potentialRepositories,
+      loaders: finalLoaders,
+      moduleService,
+      normalizedPath
+    }
+  } catch (e) {
+    logger.warn(
+      `Unable to load resources for module ${modulePath} automagically. ${e.message}`
+    )
+
+    return {} as ModuleResource
+  }
+}
+
+async function runLoaders(
+  loaders: Function[] = [],
+  {
+    localContainer,
+    container,
+    logger,
+    resolution,
+    loaderOnly,
+    registrationName,
+  }
+): Promise<void | { error: Error }> {
+  try {
+    for (const loader of loaders) {
+      await loader(
+        {
+          container: localContainer,
+          logger,
+          options: resolution.options,
+          dataLoaderOnly: loaderOnly,
+        },
+        resolution.moduleDeclaration as InternalModuleDeclaration
+      )
+    }
+  } catch (err) {
+    container.register({
+      [registrationName]: asValue(undefined),
+    })
+
+    return {
+      error: new Error(
+        `Loaders for module ${resolution.definition.label} failed: ${err.message}`
+      ),
+    }
+  }
+}
+
+function prepareLoaders({
+  loadedModuleLoaders,
+  models,
+  repositories,
+  services,
+  moduleResolution,
+  migrationPath,
+}) {
+  const finalLoaders: ModuleLoaderFunction[] = []
+
+  const toObjectReducer = (acc, curr) => {
+    acc[curr.name] = curr
+    return acc
+  }
+
+  /*
+   * If no connectionLoader function is provided, create a default connection loader.
+   * TODO: Validate naming convention
+   */
+  const connectionLoaderName = "connectionLoader"
+  const containerLoader = "containerLoader"
+
+  const hasConnectionLoader = loadedModuleLoaders.some(
+    (l) => l.name === connectionLoaderName
+  )
+
+  if (!hasConnectionLoader && models.length > 0) {
+    const connectionLoader = ModulesSdkUtils.mikroOrmConnectionLoaderFactory({
```

Comment:
would be good to get rid of MikroORM here

Reply:
This is the default connection loader, since we are using mikro orm by default. Could you elaborate on your idea?